src/Illuminate/Database/Query/Grammars/MySqlGrammar.php

```diff
@@ -166,18 +166,14 @@ protected function compileJsonUpdateColumn($key, JsonExpression $value)
      */
     public function prepareBindingsForUpdate(array $bindings, array $values)
     {
-        $index = 0;
```

Comment:
Previously we were receiving merged bindings and checked `isJsonSelector` first `count($values)` of items from bindings, because values always were first and the keys were of integer type.
In this case we can check only values and then merge with bindings and the keys can be of any type.